### DISCUSSION TITLE
Face extrusion

### DISCUSSION THREAD

**Discussion by f-pucci** [pyvista/pyvista#5011] [Q&A]
Hello everyone!

Is there a function that would let me extrude faces of a watertight geometry, while maintaining the watertightness?
If not, can you suggest any other opensource module?

I have tried to extract the faces, extrude them creating new 'solids' and then merging them with a boolean function, but the interface faces between the extrusion and the original geometry are not removed, hence making the geometry not watertight. 

I am extruding the faces at z = 0 and z= A.. very simple. but between 0 and A the geometry is quite complex (although not needed to be extruded)

**Comment by f-pucci** [comment]
Hello, anyone able to help? :(

**Comment by banesullivan** [comment]
Can you provide a graphic of what you are expecting? Something like drawing on top of a screenshot of a mesh.

I'm struggling to fully understand the task and this might help
